### PR TITLE
cleanup: docstrings for dispatcher header

### PR DESCRIPTION
**Pull request by junr03**
Description: docstrings for dispatcher header
Risk Level: low

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

File: library/common/http/dispatcher.h [hunk 1]
```diff
@@ -32,10 +32,48 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
...
+   * Send metadata over an HTTP stream. This method can be invoked multiple times.
+   * @param stream, the stream to send metadata over.
+   * @param metadata, the metadata to send.
+   * @return envoy_status_t, the resulting status of the operation.
+   */
+  envoy_status_t sendMetadata(envoy_stream_t stream, envoy_headers headers);
+
+  /**
+   * Send trailers over an open HTTP stream. This method can only be invoked once per stream.
+   * Note that this method implicitly ends the stream.
```
Comment (goaway) [inline library/common/http/dispatcher.h]: "implicitly closes (or ends) the stream locally."

File: library/common/http/dispatcher.h [hunk 2]
```diff
@@ -32,10 +32,48 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
...
+   * @param end_stream, supplies whether this is headers only.
+   * @return envoy_status_t, the resulting status of the operation.
+   */
   envoy_status_t sendHeaders(envoy_stream_t stream, envoy_headers headers, bool end_stream);
+
+  /**
+   * Send data over an open HTTP stream. This method can be invoked multiple times.
+   * @param stream, the stream to send data over.
+   * @param data, the data to send.
+   * @param end_stream, supplies whether this is the last data in the stream.
```
Comment (goaway) [inline library/common/http/dispatcher.h]: "indicates whether to close the stream locally after sending this frame"

File: library/common/http/dispatcher.h [hunk 3]
```diff
@@ -32,10 +32,48 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
...
+   * Send trailers over an open HTTP stream. This method can only be invoked once per stream.
+   * Note that this method implicitly ends the stream.
+   * @param stream, the stream to send trailers over.
+   * @param trailers, the trailers to send.
+   * @return envoy_status_t, the resulting status of the operation.
+   */
+  envoy_status_t sendTrailers(envoy_stream_t stream, envoy_headers trailers);
+
+  /**
+   * Reset an open HTTP stream.
```
Comment (rebello95) [inline library/common/http/dispatcher.h]: It'd be nice to elaborate a bit on what this means for the underlying stream handle from the consumer's perspective

File: library/common/http/dispatcher.h [hunk 4]
```diff
@@ -32,10 +32,48 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
...
+   * @return envoy_status_t, the resulting status of the operation.
+   */
   envoy_status_t sendData(envoy_stream_t stream, envoy_data data, bool end_stream);
-  envoy_status_t sendMetadata(envoy_stream_t stream, envoy_headers headers, bool end_stream);
-  envoy_status_t sendTrailers(envoy_stream_t stream, envoy_headers headers);
+
+  /**
+   * Send metadata over an HTTP stream. This method can be invoked multiple times.
+   * @param stream, the stream to send metadata over.
+   * @param metadata, the metadata to send.
```
Comment (rebello95) [inline library/common/http/dispatcher.h]: This is `header` below, but maybe it should be `metadata`?

**Comment by junr03** [comment]
@goaway @rebello95 I addressed your comments

File: library/common/http/dispatcher.h [hunk 5]
```diff
@@ -32,10 +32,49 @@ class Dispatcher : public Logger::Loggable<Logger::Id::http> {
...
+   * @return envoy_status_t, the resulting status of the operation.
+   */
+  envoy_status_t sendMetadata(envoy_stream_t stream, envoy_headers metadata);
```
Comment (rebello95) [inline library/common/http/dispatcher.h]: Does the impl also need to be updated? (Or _should_ it be updated?)

Comment (junr03) [reply library/common/http/dispatcher.h]: not implemented yet, so the cc file has no argument names, only types.

**Pull request merged**